stone/target/python_rsrc/stone_base.py

```diff
@@ -34,6 +34,16 @@ def __init__(self, tag, value=None):
         self._tag = tag
         self._value = value
 
+    def __eq__(self, other):
+        # Also need to check if one class is a subclass of another. If one union extends another,
+        # the common fields should be able to be compared to each other.
+        return (
+            (isinstance(self, other.__class__) or isinstance(other, self.__class__)) and
```

Comment:
In the simplest case of referencing the auto-populated `B.t0`, it is already a reference to `A.t0`. But, it's possible to manually instantiate `B` to `t0`, and it would take some generator work to reject that. I think out of scope for this diff. Also, the deserializer uses the manual instantiation method :( so that'll have to change.